packages/mui-joy/src/styles/ThemeProvider.tsx

```diff
@@ -1,20 +1,32 @@
 import * as React from 'react';
-import { ThemeProvider as SystemThemeProvider, useTheme as useSystemTheme } from '@mui/system';
+import {
+  ThemeProvider as SystemThemeProvider,
+  unstable_styleFunctionSx as styleFunctionSx,
+  useTheme as useSystemTheme,
+} from '@mui/system';
+import { SxProps } from './types';
 import defaultTheme, { getThemeWithVars } from './defaultTheme';
 import type { CssVarsThemeOptions } from './extendTheme';
 
 export const useTheme = () => {
   return useSystemTheme(defaultTheme);
 };
 
+const getTheme = (themeInput: CssVarsThemeOptions) => {
+  const theme = getThemeWithVars(themeInput);
+  theme.unstable_sx = (props: SxProps) => styleFunctionSx({ sx: props, theme: defaultTheme });
```

Comment:
```suggestion
  theme.unstable_sx = function (props: SxProps) {
    return styleFunctionSx({ sx: props, theme: this })
  };
```

I think using `this` makes sense in this case because when calling `theme.unstable_sx(...)`, the theme itself should be the parameter.

Reply:
Yep, this makes sense, let me just verify that it would work as expected